vpr/src/route/rr_graph_fpga_interchange.cpp

```diff
@@ -0,0 +1,1259 @@
+                            down, e_rr_type::CHANX, loc_chan_idx[std::tuple<loc_, e_rr_type>(down, e_rr_type::CHANX)],
+                            down, e_rr_type::CHANY, loc_chan_idx[std::tuple<loc_, e_rr_type>(down, e_rr_type::CHANY)]});
+                }
+            }
+
+            /* If tile has CHANY, try to cannect it to CHANY of tile below,
+             * CHANY and CHANX of tile above and CHANX of tile to the right
+             */
+            if (loc_chan_idx.find(std::tuple<loc_, e_rr_type>(loc, e_rr_type::CHANY)) != loc_chan_idx.end()) {
+                if (vertex->links[1] != nullptr && vertex->links[1]->visited == vertex) {
+                    loc_ up(std::get<0>(down), std::get<1>(down) + 1);  // CHANX
+                    loc_ up_up(std::get<0>(loc), std::get<1>(loc) + 1); // CHANY
+                    if (loc_chan_idx.find(std::tuple<loc_, e_rr_type>{up, e_rr_type::CHANX}) != loc_chan_idx.end())
+                        shorts_.push_back(std::tuple<loc_, e_rr_type, int, loc_, e_rr_type, int>{
+                            up, e_rr_type::CHANX, loc_chan_idx[std::tuple<loc_, e_rr_type>(up, e_rr_type::CHANX)],
+                            loc, e_rr_type::CHANY, loc_chan_idx[std::tuple<loc_, e_rr_type>(loc, e_rr_type::CHANY)]});
+                    if (loc_chan_idx.find(std::tuple<loc_, e_rr_type>{up_up, e_rr_type::CHANY}) != loc_chan_idx.end())
+                        shorts_.push_back(std::tuple<loc_, e_rr_type, int, loc_, e_rr_type, int>{
+                            up_up, e_rr_type::CHANY, loc_chan_idx[std::tuple<loc_, e_rr_type>(up_up, e_rr_type::CHANY)],
+                            loc, e_rr_type::CHANY, loc_chan_idx[std::tuple<loc_, e_rr_type>(loc, e_rr_type::CHANY)]});
+                }
+                if (vertex->links[2] != nullptr && vertex->links[2]->visited == vertex) {
+                    if (loc_chan_idx.find(std::tuple<loc_, e_rr_type>(right, e_rr_type::CHANX)) != loc_chan_idx.end())
+                        shorts_.push_back(std::tuple<loc_, e_rr_type, int, loc_, e_rr_type, int>{
+                            right, e_rr_type::CHANX, loc_chan_idx[std::tuple<loc_, e_rr_type>(right, e_rr_type::CHANX)],
+                            loc, e_rr_type::CHANY, loc_chan_idx[std::tuple<loc_, e_rr_type>(loc, e_rr_type::CHANY)]});
+                }
+                if (vertex->links[3] != nullptr && vertex->links[3]->visited == vertex) {
+                    if (loc_chan_idx.find(std::tuple<loc_, e_rr_type>(down, e_rr_type::CHANY)) != loc_chan_idx.end())
+                        shorts_.push_back(std::tuple<loc_, e_rr_type, int, loc_, e_rr_type, int>{
+                            down, e_rr_type::CHANY, loc_chan_idx[std::tuple<loc_, e_rr_type>(down, e_rr_type::CHANY)],
+                            loc, e_rr_type::CHANY, loc_chan_idx[std::tuple<loc_, e_rr_type>(loc, e_rr_type::CHANY)]});
+                }
+            }
+        }
+    }
+
+    /*
+     * Remove a graph represented by a root node.
+     */
+
```

Comment:
This comment might be enhanced with a description on why we should remove the graph